src/Viz.js

```diff
@@ -447,7 +462,8 @@ export default class Viz extends BaseClass {
 
         this._draw(callback);
         zoomControls.bind(this)();
-        if (this._message) this._messageClass.hide();
+
+        if (this._messageClass._isVisible && !this._noData) this._messageClass.hide();
```

Comment:
Remove all instances of `this._noData`, and just use `!this._filteredData.length` here.